iml-agent/src/daemon_plugins/snapshot.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+//! Snapshot listing plugin
+//!
+//! This module is responsible for continually discovering snapshots of filesystems.
+//!
+//!
+
+use crate::{
+    action_plugins::lustre::snapshot,
+    agent_error::ImlAgentError,
+    daemon_plugins::{DaemonPlugin, Output},
+    lustre::list_fs,
+};
+use async_trait::async_trait;
+use futures::{
+    future::{join_all, AbortHandle, Abortable},
+    lock::Mutex,
+    Future, FutureExt,
+};
+use iml_wire_types::snapshot::{List, Snapshot};
+use std::{pin::Pin, sync::Arc, time::Duration};
+use tokio::time::delay_for;
+
+struct State {
+    updated: bool,
+    output: Output,
+}
+
+#[derive(Debug)]
+pub(crate) struct SnapshotList {
+    reader: Option<AbortHandle>,
+    state: Arc<Mutex<State>>,
+}
+
+pub(crate) fn create() -> SnapshotList {
+    SnapshotList {
+        reader: None,
+        state: Arc::new(Mutex::new(State {
+            updated: false,
+            output: None,
+        })),
+    }
+}
+
+async fn list() -> Result<Vec<Snapshot>, ImlAgentError> {
+    let fss = list_fs().await?;
```

Comment:
Does this work if the MDT and MGT are on separate nodes?

Reply:
Also, does this report mounted snapshots as filesystems?

Reply:
Not, sure. But there is a problem indeed:
```
sep 02 06:38:27 mds1.local iml-agent-daemon[1882]: Sep 02 06:38:27.910 DEBUG iml_agent::daemon_plugins::snapshot: filesystems: ["02a0f396", "7cac8cc9", "zfsmo"]
sep 02 06:38:27 mds1.local iml-agent-daemon[1882]: Sep 02 06:38:27.910 DEBUG iml_cmd: Running cmd: Command { std: "/usr/sbin/lctl" "snapshot_list" "--fsname" "02a0f396", kill_on_drop: false }
sep 02 06:38:27 mds1.local iml-agent-daemon[1882]: Sep 02 06:38:27.913 DEBUG iml_cmd: Running cmd: Command { std: "/usr/sbin/lctl" "snapshot_list" "--fsname" "7cac8cc9", kill_on_drop: false }
sep 02 06:38:27 mds1.local iml-agent-daemon[1882]: Sep 02 06:38:27.915 DEBUG iml_cmd: Running cmd: Command { std: "/usr/sbin/lctl" "snapshot_list" "--fsname" "zfsmo", kill_on_drop: false }
sep 02 06:38:29 mds1.local iml-agent-daemon[1882]: Sep 02 06:38:29.597  WARN iml_agent::daemon_plugins::snapshot: listing snapshots failed: exit code: 1, stdout: , stderr: Miss MDT0 in the config file /etc/ldev.conf

```